core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1043,6 +1050,43 @@ class KafkaApis(val requestChannel: RequestChannel,
     info("Shutdown complete.")
   }
 
+  def handleCreateTopicsRequest(request: RequestChannel.Request) {
+    val createTopicsRequest = request.body.asInstanceOf[CreateTopicsRequest]
+
+    val (validTopics, duplicateTopics) = createTopicsRequest.topics.asScala.partition { case (topic, _) =>
+      !createTopicsRequest.duplicateTopics.contains(topic)
+    }
+
+    def sendResponseCallback(results: Map[String, Errors]): Unit = {
+      val respHeader = new ResponseHeader(request.header.correlationId)
+      if(duplicateTopics.nonEmpty)
+        warn(s"Create topics request from client ${request.header.clientId} contains multiple entries for the following topics: ${duplicateTopics.keySet.mkString(",")}")
+      val completeResults = results ++ duplicateTopics.keySet.map((_, Errors.INVALID_REQUEST)).toMap
+      val responseBody = new CreateTopicsResponse(completeResults.asJava)
+      trace(s"Sending create topics response $responseBody for correlation id ${request.header.correlationId} to client ${request.header.clientId}.")
+      requestChannel.sendResponse(new RequestChannel.Response(request, new ResponseSend(request.connectionId, respHeader, responseBody)))
+    }
+
+    if(!controller.isActive()) {
+      val results = createTopicsRequest.topics.asScala.map { case (topic, _) =>
+        (topic, Errors.NOT_CONTROLLER)
+      }
+      sendResponseCallback(results)
```

Comment:
Yes, in those cases, we could choose to send Errors.NOT_CONTROLLER and Errors.CLUSTER_AUTHORIZATION_FAILED back. However, since we add those duplicateTopics back with Errors.INVALID_REQUEST in sendResponseCallback(), aren't we overriding the error code for those duplicateTopics?